python/cudf/cudf/testing/parquet.py

```diff
@@ -0,0 +1,199 @@
+
+            if os.path.isfile(path) and path.endswith("_crash.json"):
+                self._load_params(path)
+            else:
+                for i in os.listdir(path):
+                    file_name = os.path.join(path, i)
+                    if os.path.isfile(file_name) and file_name.endswith(
+                        "_crash.json"
+                    ):
+                        self._load_params(file_name)
+        self._regression = True if self._inputs else False
+        self._idx = 0
+        self._current_params = {}
+
+    def _load_params(self, path):
+        with open(path, "r") as f:
+            params = json.load(f)
+        self._inputs.append(params)
+
+    @staticmethod
+    def _rand(n):
+        return random.randrange(1, n)
+
+    def generate_input(self):
+        if self._regression:
+            if self._idx >= len(self._inputs):
+                logging.info(
+                    "Reached the end of all crash.json files to run..Exiting.."
+                )
+                sys.exit(0)
+            param = self._inputs[self._idx]
+            dtypes_meta = param["dtypes_meta"]
+            num_rows = param["num_rows"]
+            num_cols = param["num_columns"]
+            seed = param["seed"]
+            random.seed(seed)
+            self._idx += 1
+            self._current_params = copy.copy(param)
+        else:
+            seed = random.randint(0, 2 ** 32 - 1)
+            random.seed(seed)
+            dtypes_meta, num_rows, num_cols = _generate_rand_meta(self)
+            self._current_params["dtypes_meta"] = dtypes_meta
+            self._current_params["seed"] = seed
+            self._current_params["num_rows"] = num_rows
+            self._current_params["num_columns"] = num_cols
+        logging.info(
+            f"Generating DataFrame with rows: {num_rows} "
+            f"and columns: {num_cols}"
+        )
+        df = cudf.DataFrame.from_arrow(
+            dg.rand_dataframe(dtypes_meta, num_rows, seed)
+        )
+        logging.info(f"Shape of DataFrame generated: {df.shape}")
+
+        return df
+
+    @property
+    def current_params(self):
+        return self._current_params
+
+
+def _generate_rand_meta(obj):
+    obj._current_params = {}
+    num_rows = obj._rand(obj._max_rows)
+    num_cols = obj._rand(obj._max_columns)
+
+    dtypes_list = list(
+        cudf.utils.dtypes.ALL_TYPES
+        - {"category", "timedelta64[ns]", "datetime64[ns]"}
+    )
+    dtypes_meta = []
+    for _ in range(num_cols):
+        dtype = random.choice(dtypes_list)
+        null_frequency = random.uniform(0, 1)
+        cardinality = obj._rand(obj._max_rows)
+        meta = dict()
+        if dtype == "str":
+            # We want to operate near the limits of string column
+            # Hence creating a string column of size almost
+            # equal to 2 Billion bytes(sizeof(int))
+            meta["max_string_length"] = 2000000000 / num_rows
```

Comment:
we don't necessarily want to always have 2B characters, just need to make sure we don't go over. I'd prefer to randomize the string len here and just limit it to the value used here.

Reply:
So this param(`max_string_length`) being set in meta will not ensure all the strings generated will be of this length. It's functionality is just the same you have mentioned, it will generate strings of random length less than value in `max_string_length`. 

This value will be fed into `schoice` api call here where this is achieved: https://github.com/rapidsai/cudf/pull/6114/files#diff-6c157a22d6751bd28a543337bd5d49f9R345-R347

Reply:
What is the distribution of string lengths? 
My concern is that we have no way to generate a table with a small number of rows and short strings. Having separate variations for rows and lengths resolves this.

Reply:
> What is the distribution of string lengths?

[0, 2000000000 / n) `n` being number of rows.



> My concern is that we have no way to generate a table with a small number of rows and short strings. Having separate variations for rows and lengths resolves this.

Okay, makes sense. In that case I'll go ahead with random number generation for `max_string_length` to ensure it is below the limits.